<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Seren Villwock" />

<meta name="date" content="2022-05-16" />

<title>Multi-Trait Simulations</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">FinalProject</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Multi-Trait Simulations</h1>
<h4 class="author">Seren Villwock</h4>
<h4 class="date">May 16, 2022</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-07-22
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>FinalProject/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges">
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted
changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges"
class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of
the R Markdown file created these results, you’ll want to first commit
it to the Git repo. If you’re still working on the analysis, you can
ignore this warning. When you’re finished, you can run
<code>wflow_publish</code> to commit the R Markdown file and build the
HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220516code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20220516)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220516code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220516)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomserenvillwockFinalProjecttree00e1b31f5978489f760dfbfd974970f829c37baetargetblank00e1b31a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/serenvillwock/FinalProject/tree/00e1b31f5978489f760dfbfd974970f829c37bae" target="_blank">00e1b31</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomserenvillwockFinalProjecttree00e1b31f5978489f760dfbfd974970f829c37baetargetblank00e1b31a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/serenvillwock/FinalProject/tree/00e1b31f5978489f760dfbfd974970f829c37bae" target="_blank">00e1b31</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  .Rprofile
    Untracked:  .gitattributes
    Untracked:  .gitignore
    Untracked:  data/DMmodel_cycle1.RDS
    Untracked:  data/DMmodel_cycle2.RDS
    Untracked:  data/DMmodel_cycle3.RDS
    Untracked:  data/MTGselection_20cycle.RDS
    Untracked:  data/MTGselection_20cycle_copy.RDS
    Untracked:  data/MTmodel_cycle1.RDS
    Untracked:  data/MTmodel_cycle2.RDS
    Untracked:  data/MTmodel_cycle3.RDS
    Untracked:  data/MTmodel_cycletest.RDS
    Untracked:  data/MTselection_20cycle.RDS
    Untracked:  data/SP.RDS
    Untracked:  data/STGselection.RDS
    Untracked:  data/STGselection_20cycle.RDS
    Untracked:  data/STGselection_copy.RDS
    Untracked:  data/STselection_20cycle.RDS
    Untracked:  data/TCmodel_cycle1.RDS
    Untracked:  data/TCmodel_cycle2.RDS
    Untracked:  data/errCov.RDS
    Untracked:  data/lasttraceG.RDS
    Untracked:  data/lasttraceP.RDS
    Untracked:  data/testsave.RDS

Unstaged changes:
    Modified:   analysis/Simulations.Rmd
    Modified:   code/calcRelationshipMatrices.R
    Modified:   code/implement_selection.R

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/Simulations.Rmd</code>) and HTML
(<code>docs/Simulations.html</code>) files. If you’ve configured a
remote Git repository (see <code>?wflow_git_remote</code>), click on the
hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/serenvillwock/FinalProject/blob/00e1b31f5978489f760dfbfd974970f829c37bae/analysis/Simulations.Rmd" target="_blank">00e1b31</a>
</td>
<td>
serenvillwock
</td>
<td>
2022-07-19
</td>
<td>
Update 7/19
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/serenvillwock/FinalProject/blob/4faffb14df6171d486cfa5511292eb4a3203f7c1/analysis/Simulations.Rmd" target="_blank">4faffb1</a>
</td>
<td>
serenvillwock
</td>
<td>
2022-06-30
</td>
<td>
fixed relationship matrix construction
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/serenvillwock/FinalProject/blob/c0e8321591cf79576dd2428febb7569b92e0b044/analysis/Simulations.Rmd" target="_blank">c0e8321</a>
</td>
<td>
serenvillwock
</td>
<td>
2022-06-30
</td>
<td>
code for generating CC matrix
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/serenvillwock/FinalProject/blob/4bcfcbe4f33255a58fd14fe4b3d13f47fd3d8d35/analysis/Simulations.Rmd" target="_blank">4bcfcbe</a>
</td>
<td>
serenvillwock
</td>
<td>
2022-06-28
</td>
<td>
initial commit
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/serenvillwock/FinalProject/4bcfcbe4f33255a58fd14fe4b3d13f47fd3d8d35/docs/Simulations.html" target="_blank">4bcfcbe</a>
</td>
<td>
serenvillwock
</td>
<td>
2022-06-28
</td>
<td>
initial commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>
<div id="objectives" class="section level2">
<h2>Objectives</h2>
<p>The objective is to determine how modeling a bivariate phenotype with
a multivariate or univariate models with either pedigree or SNP
relationship matrices will affect genetic gain with index selection.</p>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<p>I use <code>AlphaSimR</code> to simulate breeding with index
selection over 20 cycles, with simulation parameters (trait
heritabilities, covariances, etc) based on a cassava population with
negatively correlated traits, dry matter and carotenoid content (color
chart), estimated or directly from Rabbi et al. 2017. These parameters
are: DM &amp; TC additive correlation = -0.2 DM &amp; TC error
correlation = -0.1 DM error variance = 41.08 TC error variance = 4.35
Number of founders = 200 Number of progeny = 6 HistoricalNe = 200
Segregating sites = 4400 (ICGMC 2015) nQTL = 80 nSNP = 1000</p>
<p>I modeled the traits in <code>sommer</code> using either one
bivariate model incorporating both traits or two separate univariate
models, with variety as a random effect and a covariance structure based
on either additive relationships or a SNP-based genomic relationships. I
then extracted estimates of the variance components and used these to
calculate a selection index, which was applied to identify the top 1/6th
of the population with the highest economic values. These lines were
randomly crossed to generate the next generation. The gain in true
genetic values for each trait and the overall economic values were
compared after 20 cycles of selection.</p>
<div id="script-setup" class="section level3">
<h3>Script setup</h3>
<p>Install packages, start the README.md, set the random seed</p>
</div>
<div id="set-file-locations-relative-to-the-project-folder-using-here"
class="section level3">
<h3>Set file locations relative to the project folder using
<code>here</code></h3>
<pre class="r"><code>here::i_am(&quot;analysis/Simulations.Rmd&quot;)</code></pre>
</div>
<div id="generate-a-founder-population" class="section level3">
<h3>Generate a founder population</h3>
<pre class="r"><code>source(&quot;code/implement_selection.R&quot;)
progeny &lt;- generate_founders()
saveRDS(progeny, file = &quot;data/founder_progeny_pop.RDS&quot;)</code></pre>
<pre class="r"><code># Load founder population
progeny &lt;- readRDS(&quot;data/founder_progeny_pop.RDS&quot;)
SP &lt;- readRDS(&quot;data/SP.RDS&quot;)
errCov &lt;- readRDS(&quot;data/errCov.RDS&quot;)</code></pre>
</div>
<div id="implement-phenotypic-selection" class="section level3">
<h3>Implement phenotypic selection</h3>
</div>
<div id="set-economic-weights" class="section level3">
<h3>Set economic weights</h3>
<pre class="r"><code>DMval = 1
TCval = 1</code></pre>
<pre class="r"><code>source(&quot;code/implement_selection.R&quot;)

MTselection &lt;- implement_pheno_selection(progeny, multitrait = T, nCycles = 20)

saveRDS(MTselection, file = &quot;data/MTselection_20cycle_B.RDS&quot;)</code></pre>
<pre class="r"><code>source(&quot;code/implement_selection.R&quot;)

STselection &lt;- implement_pheno_selection(progeny, multitrait = F, nCycles = 20)

saveRDS(STselection, file = &quot;data/STselection_20cycle_B.RDS&quot;)</code></pre>
<pre class="r"><code>source(&quot;code/implement_selection.R&quot;)

MTGselection &lt;- implement_g_selection(progeny, multitrait = F, nCycles = 20)

saveRDS(MTselection, file = &quot;data/MTGselection_20cycle_B.RDS&quot;)</code></pre>
<pre class="r"><code>source(&quot;code/implement_selection.R&quot;)

STGselection &lt;- implement_g_selection(progeny, multitrait = F, nCycles = 20)

saveRDS(STGselection, file = &quot;data/STGselection_20cycle_B.RDS&quot;)</code></pre>
<p>Reload data</p>
<pre class="r"><code>MTselection &lt;- readRDS(file = &quot;./data/MTselection_20cycle.RDS&quot;)
STselection &lt;- readRDS(file = &quot;./data/STselection_20cycle.RDS&quot;)
MTGselection &lt;- readRDS(file = &quot;./data/MTGselection_20cycle.RDS&quot;)
STGselection &lt;- readRDS(file = &quot;./data/STGselection_20cycle.RDS&quot;)


# Set variable types
MTselection$DM_gv &lt;- as.numeric(MTselection$DM_gv)
MTselection$TC_gv &lt;- as.numeric(MTselection$TC_gv)

STselection$DM_gv &lt;- as.numeric(STselection$DM_gv)
STselection$TC_gv &lt;- as.numeric(STselection$TC_gv)

MTGselection$DM_gv &lt;- as.numeric(MTGselection$DM_gv)
MTGselection$TC_gv &lt;- as.numeric(MTGselection$TC_gv)

STGselection$DM_gv &lt;- as.numeric(STGselection$DM_gv)
STGselection$TC_gv &lt;- as.numeric(STGselection$TC_gv)

# Calculate overall economic value
MTselection$value &lt;- scale(MTselection$DM_gv) + scale(MTselection$TC_gv)
STselection$value &lt;- scale(STselection$DM_gv) + scale(STselection$TC_gv)
MTGselection$value &lt;- scale(MTGselection$DM_gv) + scale(MTGselection$TC_gv)
STGselection$value &lt;- scale(STGselection$DM_gv) + scale(STGselection$TC_gv)</code></pre>
</div>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<p>For dry matter, the multi-trait model with the additive relationship
matrix performed the best, and the other models performed similarly. The
gains for carotenoid content were all over the place because the scaling
of the traits needs to be fixed so it is in the same units as dry matter
(dry matter, with larger units, unintentionally received most of the
weight in the selection index). While the multi-trait model with the
genomic relationship matrix performed best in this context, and the
multi-trait model with the additive relationship matrix performed the
worst, I imagine that it may change when the scaling is fixed.</p>
<pre class="r"><code>#plot progress for dry matter
ggplot(MTselection, aes(cycle, DM_gv)) + geom_boxplot(data = MTselection, aes(as.factor(cycle),
    DM_gv, color = &quot;multi-trait model with A&quot;)) + geom_boxplot(data = STselection,
    aes(as.factor(cycle), DM_gv, color = &quot;single-trait models with A&quot;)) + geom_boxplot(data = MTGselection,
    aes(as.factor(cycle), DM_gv, color = &quot;multi-trait model with G&quot;)) + geom_boxplot(data = STGselection,
    aes(as.factor(cycle), DM_gv, color = &quot;single-trait models with G&quot;)) + xlab(&quot;cycle of selection&quot;) +
    ylab(&quot;dry matter genetic value&quot;) + scale_color_discrete(&quot;selection method&quot;) +
    labs(title = &quot;selection for dry matter&quot;)

#plot progress for total carotenoids
ggplot(MTselection, aes(cycle, TC_gv)) + geom_boxplot(data = MTselection, aes(as.factor(cycle),
    TC_gv, color = &quot;multi-trait model with A&quot;)) + geom_boxplot(data = STselection,
    aes(as.factor(cycle), TC_gv, color = &quot;single-trait models with A&quot;)) + geom_boxplot(data = STGselection,
    aes(as.factor(cycle), TC_gv, color = &quot;single-trait models with G&quot;)) + geom_boxplot(data = MTGselection,
    aes(as.factor(cycle), TC_gv, color = &quot;multi-trait model with G&quot;)) + xlab(&quot;cycle of selection&quot;) +
    ylab(&quot;total carotenoids genetic value&quot;) + scale_color_discrete(&quot;selection method&quot;) +
    labs(title = &quot;selection for carotenoids&quot;)


#plot progress for dry matter with G matrix
ggplot(MTGselection, aes(cycle, DM_gv)) + geom_boxplot(data = MTGselection, aes(as.factor(cycle),
    DM_gv, color = &quot;multi-trait model&quot;)) + geom_boxplot(data = STGselection, aes(as.factor(cycle),
    DM_gv, color = &quot;single-trait models&quot;)) + xlab(&quot;cycle of selection&quot;) + ylab(&quot;dry matter genetic value&quot;) +
    scale_color_discrete(&quot;selection method&quot;) + labs(title = expression(atop(bold(&quot;selection for dry matter&quot;),
    atop(bold(&quot;using genomic relationship matrix&quot;), &quot;&quot;))))


#plot progress for total carotenoids 
ggplot(MTGselection, aes(cycle, TC_gv)) + geom_boxplot(data = MTGselection, aes(as.factor(cycle),
    TC_gv, color = &quot;multi-trait model&quot;)) + geom_boxplot(data = STGselection, aes(as.factor(cycle),
    TC_gv, color = &quot;single-trait models&quot;)) + xlab(&quot;cycle of selection&quot;) + ylab(&quot;total carotenoids genetic value&quot;) +
    scale_color_discrete(&quot;selection method&quot;) + labs(title = expression(atop(bold(&quot;selection for carotenoids&quot;),
    atop(bold(&quot;using genomic relationship matrix&quot;), &quot;&quot;))))




#plot progress for overall economic value 
ggplot(MTGselection, aes(cycle, value)) + geom_boxplot(data = MTGselection, aes(as.factor(cycle),
    value, color = &quot;multi-trait model, G&quot;)) + geom_boxplot(data = STGselection, aes(as.factor(cycle),
    value, color = &quot;single-trait models, G&quot;)) + geom_boxplot(data = MTselection,
    aes(as.factor(cycle), value, color = &quot;multi-trait model, A&quot;)) + geom_boxplot(data = STselection,
    aes(as.factor(cycle), value, color = &quot;single-trait models, A&quot;)) + xlab(&quot;cycle of selection&quot;) +
    ylab(&quot;total economic value&quot;) + scale_color_discrete(&quot;selection method&quot;) + labs(title = expression(atop(bold(&quot;gain in economic value&quot;),
    &quot;&quot;, &quot;&quot;)))</code></pre>
<p><img src="figure/Simulations.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/Simulations.Rmd/unnamed-chunk-5-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/Simulations.Rmd/unnamed-chunk-5-3.png" width="672" style="display: block; margin: auto;" /><img src="figure/Simulations.Rmd/unnamed-chunk-5-4.png" width="672" style="display: block; margin: auto;" /><img src="figure/Simulations.Rmd/unnamed-chunk-5-5.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="discussion" class="section level2">
<h2>Discussion</h2>
<p>With just one simulation run of the 20 cycles, I can’t be sure yet if
these results are systematic or stochastic. I also need to fix the
scaling of the trait values so that they are not just centered around 0
but also scaled to a standard deviation of 1. For this simulation, I
used a negative error correlation between TC and DM, since the additive
genetic correlation is less negative than the phenotypic correlation
between the traits. Theoretically, I think that the multi-trait genomic
relationship models may have an advantage because it can inform which
varieties have variants that minimize the trade-off between traits
(under a model of pleiotropy).</p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
R version 4.1.2 (2021-11-01)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] here_1.0.1      sommer_4.1.7    crayon_1.5.1    lattice_0.20-45
 [5] MASS_7.3-54     Matrix_1.3-4    devtools_2.4.3  usethis_2.1.5  
 [9] forcats_0.5.1   stringr_1.4.0   dplyr_1.0.8     purrr_0.3.4    
[13] readr_2.1.2     tidyr_1.2.0     tibble_3.1.6    ggplot2_3.3.5  
[17] tidyverse_1.3.1 AlphaSimR_1.0.4 R6_2.5.1       

loaded via a namespace (and not attached):
 [1] fs_1.5.2          lubridate_1.8.0   httr_1.4.2        rprojroot_2.0.2  
 [5] tools_4.1.2       backports_1.4.1   utf8_1.2.2        DBI_1.1.2        
 [9] colorspace_2.0-2  withr_2.4.3       tidyselect_1.1.1  prettyunits_1.1.1
[13] processx_3.5.2    compiler_4.1.2    git2r_0.29.0      cli_3.2.0        
[17] rvest_1.0.2       formatR_1.11      xml2_1.3.3        desc_1.4.0       
[21] labeling_0.4.2    scales_1.1.1      callr_3.7.0       digest_0.6.29    
[25] rmarkdown_2.11    pkgconfig_2.0.3   htmltools_0.5.2   sessioninfo_1.2.2
[29] highr_0.9         dbplyr_2.1.1      fastmap_1.1.0     rlang_1.0.1      
[33] readxl_1.3.1      rstudioapi_0.13   farver_2.1.0      jquerylib_0.1.4  
[37] generics_0.1.2    jsonlite_1.7.3    magrittr_2.0.2    Rcpp_1.0.8.3     
[41] munsell_0.5.0     fansi_1.0.2       lifecycle_1.0.1   stringi_1.7.6    
[45] whisker_0.4       yaml_2.2.2        brio_1.1.3        pkgbuild_1.3.1   
[49] grid_4.1.2        promises_1.2.0.1  haven_2.4.3       hms_1.1.1        
[53] knitr_1.37        ps_1.6.0          pillar_1.7.0      pkgload_1.2.4    
[57] reprex_2.0.1      glue_1.6.1        evaluate_0.14     remotes_2.4.2    
[61] modelr_0.1.8      vctrs_0.3.8       tzdb_0.2.0        httpuv_1.6.5     
[65] testthat_3.1.2    cellranger_1.1.0  gtable_0.3.0      assertthat_0.2.1 
[69] cachem_1.0.6      xfun_0.29         broom_0.7.12      later_1.3.0      
[73] memoise_2.0.1     workflowr_1.7.0   ellipsis_0.3.2   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
