---
title: "Multi-Trait Simulations"
author: "Seren Villwock"
date: "May 16, 2022"
output:
  html_document:
    fig_caption: yes
    fig_height: 6
    fig_width: 12
    highlight: haddock
    theme: cosmo
    toc: yes
    toc_float:
      collapsed: no
      toc_depth: 2
  pdf_document:
    toc: yes
editor_options:
  chunk_output_type: console
---
  
<style>
pre {
  overflow-x: auto;
}
pre code {
  word-wrap: normal;
  white-space: pre;
}
</style>

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo=TRUE)
knitr::opts_chunk$set(message=FALSE)
knitr::opts_chunk$set(cache=FALSE)
knitr::opts_chunk$set(tidy=TRUE)
knitr::opts_chunk$set(results="hold")
knitr::opts_chunk$set(collapse=TRUE)
knitr::opts_chunk$set(fig.show="hold")
knitr::opts_chunk$set(tidy.opts = list(wrap=FALSE))
```


## Objectives
The objective is to determine how modeling a bivariate phenotype with a multivariate or univariate models with either pedigree or SNP relationship matrices will affect genetic gain with index selection. 


## Methods
I use `AlphaSimR` to simulate breeding with index selection over 20 cycles, with simulation parameters (trait heritabilities, covariances, etc) based on a cassava population with negatively correlated traits, dry matter and carotenoid content (color chart), estimated or directly from Rabbi et al. 2017. These parameters are:
DM & TC additive correlation = -0.2
DM & TC error correlation = -0.1
DM error variance = 41.08
TC error variance = 4.35
Number of founders = 200
Number of progeny = 6
HistoricalNe = 200
Segregating sites = 4400 (ICGMC 2015)
nQTL = 80
nSNP = 1000

I modeled the traits in `sommer` using either one bivariate model incorporating both traits or two separate univariate models, with variety as a random effect and a covariance structure based on either additive relationships or a SNP-based genomic relationships. I then extracted estimates of the variance components and used these to calculate a selection index, which was applied to identify the top 1/6th of the population with the highest economic values. These lines were randomly crossed to generate the next generation. The gain in true genetic values for each trait and the overall economic values were compared after 20 cycles of selection.



### Script setup
Install packages, start the README.md, set the random seed  
```{r Script setup, echo=FALSE}
ip <- installed.packages()
packages_used <- c("AlphaSimR", "tidyverse", "devtools","sommer","here", "ggplot2")

for (package in packages_used){
  isInstalled <- package %in% rownames(ip)
  if (!isInstalled){
    stop(paste("ERROR: you need to install the", package, "package"))
  } else{
    library(package, character.only=T)
  }
}#END packages_used

random_seed <- 571997
set.seed(random_seed)
```


### Set file locations relative to the project folder using `here`  
```{r set here path}
here::i_am("analysis/Simulations.Rmd")
```


### Generate a founder population
```{r, eval=F}
source("code/implement_selection.R")
progeny <- generate_founders()
saveRDS(progeny, file="data/founder_progeny_pop.RDS")
```

```{r}
# Load founder population
progeny <- readRDS("data/founder_progeny_pop.RDS")
SP <- readRDS("data/SP.RDS")
errCov <- readRDS("data/errCov.RDS")
```


### Implement phenotypic selection
### Set economic weights
```{r}
DMval = 1
TCval = 1
```

```{r, multitrait model selection w/ Amat, eval=F}
source("code/implement_selection.R")

MTselection <- implement_pheno_selection(progeny, multitrait=T, nCycles=20) 

saveRDS(MTselection, file="data/MTselection_20cycle_B.RDS")
```

```{r, single trait models selection w/ Amat, eval=F}
source("code/implement_selection.R")

STselection <- implement_pheno_selection(progeny, multitrait=F, nCycles=20) 

saveRDS(STselection, file="data/STselection_20cycle_B.RDS")
```

```{r, multi trait model selection w/ Gmat, eval=F}
source("code/implement_selection.R")

MTGselection <- implement_g_selection(progeny, multitrait=F, nCycles=20) 

saveRDS(MTselection, file="data/MTGselection_20cycle_B.RDS")
```

```{r, single trait models phenotypic selection w/ Gmat, eval=F}
source("code/implement_selection.R")

STGselection <- implement_g_selection(progeny, multitrait=F, nCycles=20) 

saveRDS(STGselection, file="data/STGselection_20cycle_B.RDS")
```

Reload data 
```{r}
MTselection <- readRDS(file="./data/MTselection_20cycle.RDS")
STselection <- readRDS(file="./data/STselection_20cycle.RDS")
MTGselection <- readRDS(file="./data/MTGselection_20cycle.RDS")
STGselection <- readRDS(file="./data/STGselection_20cycle.RDS")


# Set variable types
MTselection$DM_gv <- as.numeric(MTselection$DM_gv)
MTselection$TC_gv <- as.numeric(MTselection$TC_gv)

STselection$DM_gv <- as.numeric(STselection$DM_gv)
STselection$TC_gv <- as.numeric(STselection$TC_gv)

MTGselection$DM_gv <- as.numeric(MTGselection$DM_gv)
MTGselection$TC_gv <- as.numeric(MTGselection$TC_gv)

STGselection$DM_gv <- as.numeric(STGselection$DM_gv)
STGselection$TC_gv <- as.numeric(STGselection$TC_gv)

# Calculate overall economic value
MTselection$value <- scale(MTselection$DM_gv) + scale(MTselection$TC_gv) 
STselection$value <- scale(STselection$DM_gv) + scale(STselection$TC_gv) 
MTGselection$value <- scale(MTGselection$DM_gv) + scale(MTGselection$TC_gv) 
STGselection$value <- scale(STGselection$DM_gv) + scale(STGselection$TC_gv) 
```



## Results

For dry matter, the multi-trait model with the additive relationship matrix performed the best, and the other models performed similarly. The gains for carotenoid content were all over the place because the scaling of the traits needs to be fixed so it is in the same units as dry matter (dry matter, with larger units, unintentionally received most of the weight in the selection index). While the multi-trait model with the genomic relationship matrix performed best in this context, and the multi-trait model with the additive relationship matrix performed the worst, I imagine that it may change when the scaling is fixed. 

```{r}
#plot progress for dry matter
ggplot(MTselection, aes(cycle, DM_gv)) +
  geom_boxplot(data= MTselection, aes(as.factor(cycle), DM_gv, color= "multi-trait model with A")) +
  geom_boxplot(data= STselection, aes(as.factor(cycle), DM_gv, color="single-trait models with A")) +
  geom_boxplot(data= MTGselection, aes(as.factor(cycle), DM_gv, color="multi-trait model with G")) +
  geom_boxplot(data= STGselection, aes(as.factor(cycle), DM_gv, color="single-trait models with G")) +
  xlab("cycle of selection") +
  ylab("dry matter genetic value") +
  scale_color_discrete("selection method") +
  labs(title = "selection for dry matter")

#plot progress for total carotenoids
ggplot(MTselection, aes(cycle, TC_gv)) +
  geom_boxplot(data= MTselection, aes(as.factor(cycle), TC_gv, color= "multi-trait model with A")) +
  geom_boxplot(data= STselection, aes(as.factor(cycle), TC_gv, color="single-trait models with A")) +
  geom_boxplot(data= STGselection, aes(as.factor(cycle), TC_gv, color="single-trait models with G")) +
  geom_boxplot(data= MTGselection, aes(as.factor(cycle), TC_gv, color="multi-trait model with G")) +
  xlab("cycle of selection") +
  ylab("total carotenoids genetic value") +
  scale_color_discrete("selection method") +
  labs(title = "selection for carotenoids")


#plot progress for dry matter with G matrix
ggplot(MTGselection, aes(cycle, DM_gv)) +
  geom_boxplot(data= MTGselection, aes(as.factor(cycle), DM_gv, color= "multi-trait model")) +
  geom_boxplot(data= STGselection, aes(as.factor(cycle), DM_gv, color="single-trait models")) +
  xlab("cycle of selection") +
  ylab("dry matter genetic value") +
  scale_color_discrete("selection method") +
  labs(title = expression(atop(bold("selection for dry matter"), 
                             atop(bold("using genomic relationship matrix"), ""))))
  
  
#plot progress for total carotenoids 
ggplot(MTGselection, aes(cycle, TC_gv)) +
  geom_boxplot(data= MTGselection, aes(as.factor(cycle), TC_gv, color= "multi-trait model")) +
  geom_boxplot(data= STGselection, aes(as.factor(cycle), TC_gv, color="single-trait models")) +
  xlab("cycle of selection") +
  ylab("total carotenoids genetic value") +
  scale_color_discrete("selection method") +
  labs(title = expression(atop(bold("selection for carotenoids"), 
                             atop(bold("using genomic relationship matrix"), ""))))




#plot progress for overall economic value 
ggplot(MTGselection, aes(cycle, value)) +
  geom_boxplot(data= MTGselection, aes(as.factor(cycle), value, color= "multi-trait model, G")) +
  geom_boxplot(data= STGselection, aes(as.factor(cycle), value, color="single-trait models, G")) +
  geom_boxplot(data= MTselection, aes(as.factor(cycle), value, color= "multi-trait model, A")) +
  geom_boxplot(data= STselection, aes(as.factor(cycle), value, color="single-trait models, A")) +
  xlab("cycle of selection") +
  ylab("total economic value") +
  scale_color_discrete("selection method") +
  labs(title = expression(atop(bold("gain in economic value"),"", "")))

```

## Discussion
With just one simulation run of the 20 cycles, I can't be sure yet if these results are systematic or stochastic. I also need to fix the scaling of the trait values so that they are not just centered around 0 but also scaled to a standard deviation of 1. For this simulation, I used a negative error correlation between TC and DM, since the additive genetic correlation is less negative than the phenotypic correlation between the traits. Theoretically, I think that the multi-trait genomic relationship models may have an advantage because it can inform which varieties have variants that minimize the trade-off between traits (under a model of pleiotropy).
